pyro/infer/mcmc/mcmc.py

```diff
@@ -275,12 +275,15 @@ class MCMCMarginals(Marginals):
     def diagnostics(self):
         if self._diagnostics:
             return self._diagnostics
+        support = self.support()
         for site in self.sites:
+            if self._trace_posterior.num_chains == 1:
+                site_support = support[site].unsqueeze(0)
```

Comment:
Bug: when `num_chains=1`, we have to create a new dimension to calculate diagnostics statistics.